primeqa/qg/processors/hybrid_qg/hybridqa_processor.py

```diff
@@ -0,0 +1,252 @@
+import json
+import numpy as np
+from copy import deepcopy
+
+from typing import Dict
+from dataclasses import dataclass
+
+import stanza
+from tqdm import tqdm
+from datasets import Dataset, load_dataset
+from primeqa.qg.utils.constants import QGSpecialTokens
+
+from transformers import PreTrainedTokenizer
+
+from sklearn.metrics.pairwise import linear_kernel
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+@dataclass
+class HybridQAProcessor():
+    def __init__(self, tokenizer=None, input_max_len=512, target_max_len=20):
+        """
+        Class for sampling hybrid chains from a table and its linked passages, to be used for HybridQG training.
+        Args:
+            dataset_name (str): Supports only hybrid_qa for now.
+        """
+        self.tokenizer = tokenizer
+        self.input_max_len = input_max_len
+        self.target_max_len = target_max_len
+        self.nlp_model = stanza.Pipeline(lang='en', processors='tokenize', verbose=False)
+        self.answer_separator = " %s " % (QGSpecialTokens.ans)
+        self.hop_separator = " %s " % (QGSpecialTokens.hsep)
+        self.meta_separator = " %s " % (QGSpecialTokens.header)
+
+    def __call__(self, dataset) -> Dataset:
```

Comment:
Can you add more docstring comments to this file? Also in some cases there are comments but they are not docstrings

Reply:
Done